packages/framework/tests/Unit/Component/Money/MoneyTest.php

```diff
@@ -0,0 +1,582 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrameworkBundle\Unit\Component\Money;
+
+use Iterator;
+use Litipk\BigNumbers\Errors\BigNumbersError;
+use PHPUnit\Framework\TestCase;
+use Shopsys\FrameworkBundle\Component\Money\Money;
+
+class MoneyTest extends TestCase
+{
+    /**
+     * @dataProvider fromStringProvider
+     * @param string $string
+     * @param string $expected
+     */
+    public function testFromString(string $string, string $expected): void
+    {
+        $money = Money::fromString($string);
+
+        $this->assertEquals($expected, $money->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function fromStringProvider(): Iterator
+    {
+        yield ['0', '0'];
+        yield ['-0', '-0'];
```

Comment:
I've added the test case because it surprised me (but is somewhat understandable) and I wanted to know if the behavior changed in the future. Maybe we'd like to change this behavior. I just wanted to be explicit about it.

Reply:
After the weekend without using my brain, I'm refreshed and back :smiling_imp:

Reply:
We have added some new test cases to test this scenario and we have updated Money object to handle this. We have found some unexpected behavior we had to resolve. More info: c0673b8f0082a79babb3ae01eaa95f403008ba7a

Reply:
I'd like to clarify that zero still has multiple (meaningful) representations due to its scale: `0`, `0.0`, `0.00`, ...

Reply:
Does it really have? Is `0.000 == 0`? If so, then the representation is not meaningful in my opinion. Because `0.00` only expresses that it has `scale 2`, but it is still the same value.